plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/InstallManagedCloudSdkStartup.java

```diff
@@ -0,0 +1,28 @@
+package com.google.cloud.tools.eclipse.sdk.ui;
+
+import com.google.cloud.tools.eclipse.sdk.CloudSdkManager;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.ui.IStartup;
+import org.eclipse.ui.progress.WorkbenchJob;
+
+/** A simple startup task that triggers installation of the Google Cloud SDK if applicable. */
+public class InstallManagedCloudSdkStartup implements IStartup {
+
+  @Override
+  public void earlyStartup() {
+    // Use a WorkbenchJob in case the user decides to exit quickly
+    Job triggerInstallationJob =
+        new WorkbenchJob("Check Google Cloud SDK") {
+          @Override
+          public IStatus runInUIThread(IProgressMonitor monitor) {
```

Comment:
Remind me, why does this need to run in the UI thread? My Eclipse thread-fu is weak. :-)

Reply:
earlyStartup() is called on a non-UI worker thread and runs before the UI is actually rendered.  Since we're about to do something relatively expensive, we should wait until the workbench is actually up and running, especially so we don't slow down startup.

(I'll add a comment to this effect.)

(Maybe I should increase the `schedule()` to like 10 seconds.)